MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InputController.h

```diff
@@ -245,29 +245,59 @@ class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerDraw
 /**
     Controller for erasing peaks on an unwrapped surface.
   */
-class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerErase
+class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerSelection
     : public InputControllerDraw {
   Q_OBJECT
 
 public:
-  InputControllerErase(QObject *parent);
-  ~InputControllerErase() override;
+  InputControllerSelection(QObject *parent);
+  ~InputControllerSelection() override;
   void onPaint(QPainter &) override;
 
-signals:
-  void erase(const QRect &);
-
 private:
   void drawCursor(QPixmap *cursor) override;
-  void signalLeftClick() override;
   void setPosition(const QPoint &pos) override;
   void resize() override;
 
-  QRect m_rect;
+protected:
```

Comment:
1) Yes that is possible to add a constructor for that, in that case it might actually be easier to just have one class instead of base + 2 derived classes. They do the same thing but use different signals/images.

2) Dependant on the answer to 1 anyway, but also I don't see where the duplicate member name is?